packages/mask/src/plugins/RealityCards/SNSAdaptor/CardDialog.tsx

```diff
@@ -0,0 +1,445 @@
+        flex: 1,
+        textAlign: 'left',
+    },
+    message: {
+        textAlign: 'center',
+    },
+    refresh: {
+        bottom: theme.spacing(1),
+        right: theme.spacing(1),
+        fontSize: 'inherit',
+    },
+}))
+
+interface InputBoxProps {
+    label: string
+    value: string
+    onChange: (value: string) => void
+}
+
+function InputBox(props: InputBoxProps) {
+    const { label, value, onChange } = props
+
+    return (
+        <FormControl sx={{ m: 0.5 }} variant="outlined">
+            <OutlinedInput
+                id="outlined-adornment-weight"
+                type=""
+                value={value}
+                onChange={(e) => {
+                    e.target.value.match(/^\d*$/) ? onChange(e.target.value) : null
+                }}
+                endAdornment={<InputAdornment position="end">{label}</InputAdornment>}
+                aria-describedby="outlined-weight-helper-text"
+                placeholder="0"
+                inputProps={{
+                    'aria-label': 'weight',
+                    autoComplete: 'off',
+                    autoCorrect: 'off',
+                    inputMode: 'decimal',
+                    spellCheck: false,
+                }}
+            />
+        </FormControl>
+    )
+}
+
+interface CardDialogProps {
+    open: boolean
+    onClose: () => void
+    market: Market
+    card: Card
+}
+
+export function CardDialog(props: CardDialogProps) {
+    const { open, onClose, card, market } = props
+    const { t } = useI18N()
+    const { classes } = useStyles()
+
+    // #region context
+    const account = useAccount()
+    const token = useBaseToken()
+
+    const [depositDialogOpen, setDepositDialogOpen] = useState(false)
+    const [inputPrice, setInputPrice] = useState('')
+    const [duration, setDuration] = useState(0)
+    const [limitedOwnership, setLimitedOwnership] = useState(false)
+    const [minutes, setMinutes] = useState('')
+    const [hours, setHours] = useState('')
+    const [days, setDays] = useState('')
+
+    const isOwner = useMemo(() => {
+        return isSameAddress(card.owner?.id, account)
+    }, [card.owner?.id, account])
+    // #endregion
+
+    useEffect(() => {
+        setDuration(
+            limitedOwnership
+                ? Number.parseInt(minutes || '0', 10) * 60 +
+                      Number.parseInt(hours || '0', 10) * 60 * 60 +
+                      Number.parseInt(days || '0', 10) * 60 * 60 * 24
+                : 0,
+        )
+    }, [days, hours, minutes, limitedOwnership])
+
+    const onDialogClose = () => {
+        setInputPrice('')
+        onClose()
+    }
+
+    // #region balance
+    const {
+        value: tokenBalance = '0',
+        loading: loadingTokenBalance,
+        error: errorTokenBalance,
+        retry: tokenBalanceRetry,
+    } = useUserDeposit()
+    // #endregion
+
+    // #region priceHourly
+    const priceHourly = new BigNumber(formatAmount(new BigNumber(inputPrice ?? 0), token.decimals))
```

Comment:
The `formatAmount` is only for the user interface, don't use it as part of internal calculation.

Reply:
Ops did not know that, replacing all of the occurrences with `BigNumber.shiftedBy`.